manifests/integrations/postgres.pp

```diff
@@ -74,7 +75,25 @@
     $dst = "${datadog_agent::conf_dir}/postgres.yaml"
   }
 
-  file { $dst:
+  if !$instances and $host {
+    $_instances = [{
+      'host'           => $host,
+      'password'       => $password,
+      'dbname'         => $dbname,
+      'port'           => $port,
+      'username'       => $username,
+      'use_psycopg2'   => $use_psycopg2,
+      'tags'           => $tags,
+      'tables'         => $tables,
+      'custom_metrics' => $custom_metrics,
+    }]
+  } elsif !$instances{
+    $_instances = []
+  } else {
+    $_instances = $instances
+  }
+
+  file { "$dst":
```

Comment:
This line has a minor stylistic issue:
```
manifests/integrations/postgres.pp:96:only_variable_string:WARNING:string containing only a variable
manifests/integrations/postgres.pp:96:variables_not_enclosed:WARNING:variable not enclosed in {}
```

The line was originally:
```
  file { $dst:
```
which I think is fine, but if you have your reasons for changing that you might have to address it some other way.